src/agentscope/models/litellm_model.py

```diff
@@ -0,0 +1,237 @@
+# -*- coding: utf-8 -*-
+"""Model wrapper based on litellm https://docs.litellm.ai/docs/"""
+from abc import ABC
+from typing import Union, Any, List, Sequence
+import os
+
+from loguru import logger
+
+from .model import ModelWrapperBase, ModelResponse
+from ..message import MessageBase
+from ..utils.tools import _convert_to_str
+
+try:
+    import litellm
+except ImportError:
+    litellm = None
+
+
+class LiteLLMWrapperBase(ModelWrapperBase, ABC):
+    """The model wrapper based on LiteLLM API."""
+
+    def __init__(
+        self,
+        config_name: str,
+        model_name: str = None,
+        api_key: str = None,
+        api_key_name: str = None,
+        generate_args: dict = None,
+        **kwargs: Any,
+    ) -> None:
+        """
+        Args:
+            config_name (`str`):
+                The name of the model config.
+            model_name (`str`, default `None`):
+                The name of the model to use in OpenAI API.
+            api_key (`str`, default `None`):
+                The API key used.
+            api_key_name (`str`, default `None`):
+                The API key name used, related to the model_name.
+            generate_args (`dict`, default `None`):
+                The extra keyword arguments used in litellm api generation,
+                e.g. `temperature`, `seed`.
+
+        """
+
+        if model_name is None:
+            model_name = config_name
+            logger.warning("model_name is not set, use config_name instead.")
+
+        super().__init__(config_name=config_name)
+
+        if litellm is None:
+            raise ImportError(
+                "Cannot find litellm package in current python environment.",
+            )
+
+        self.model_name = model_name
+        self.generate_args = generate_args or {}
+        self.api_key = api_key
+        self.api_key_name = api_key_name
```

Comment:
One questions is that should we provide some hints for this api_key_name? 
Or we remove api_key and api_key_name arguments, and default that all api_key are obtained from environment.